realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/SessionTests.kt

```diff
@@ -0,0 +1,562 @@
+/*
+ * Copyright 2016 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm
+
+import androidx.test.annotation.UiThreadTest
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import io.realm.TestHelper.TestLogger
+import io.realm.entities.StringOnly
+import io.realm.entities.StringOnlyModule
+import io.realm.exceptions.RealmFileException
+import io.realm.exceptions.RealmMigrationNeededException
+import io.realm.log.RealmLog
+import io.realm.rule.BlockingLooperThread
+import io.realm.util.ResourceContainer
+import org.hamcrest.CoreMatchers
+import org.hamcrest.MatcherAssert.assertThat
+import org.junit.*
+import org.junit.runner.RunWith
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.atomic.AtomicBoolean
+import kotlin.test.assertEquals
+import kotlin.test.assertFalse
+import kotlin.test.assertTrue
+import kotlin.test.fail
+
+@RunWith(AndroidJUnit4::class)
+class SessionTests {
+    private lateinit var configuration: SyncConfiguration
+    private lateinit var app: TestRealmApp
+    private lateinit var user: RealmUser
+
+    @get:Rule
+    val configFactory = TestSyncConfigurationFactory()
+
+    private val looperThread = BlockingLooperThread()
+
+    @Before
+    fun setUp() {
+        Realm.init(InstrumentationRegistry.getInstrumentation().targetContext)
+        app = TestRealmApp()
+        // TODO We could potentially work without a fully functioning user to speed up tests, but
+        //  seems like the old  way of "faking" it, does now work for now, so using a real user.
+        // user = SyncTestUtils.createTestUser(app)
+        user = createNewUser(app)
+        configuration = SyncConfiguration.defaultConfig(user, "default")
+    }
+
+    @After
+    fun tearDown() {
+        if (this::app.isInitialized) {
+            app.close()
+        }
+    }
+
+    @Test
+    fun get_syncValues() {
+        val session = SyncSession(configuration)
```

Comment:
It should not be possible to create SyncSessions from the public API.... you need to either fetch it using :
```
realm.syncSession // Kotlin extension
app.sync.getSession(syncConfig) // Java API
```

Both require opening a Realm.

But maybe we did this here and in other tests because it is a lot easier.

Reply:
I updated all session to use this approach.